pkg/api/controller.go

```diff
@@ -175,7 +179,9 @@ func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body LoginJSO
 	secret := c.Auth.SecretStore().SharedSecret()
 	// user.Username will be different from username/access_key_id on
 	// LDAP login.  Use the stored value.
-	tokenString, err := GenerateJWT(secret, user.ID, loginTime, expires)
+	// Audience and email set here to be an empty string for backward compatibility
+	tokenString, err :=
+		GenerateJWT(secret, LoginAudience, user.ID, *user.Email, loginTime, expires)
```

Comment:
I don't understand how this works.  Say I have a user with no email.  I would expect to receive `user.Email == nil`.  And when that user logs in with their access key (supported!), does this panic?

Reply:
Good point. In any case, as @nopcoder suggested, the JWT creation function was split into two separate functions, so the email will not be passed in any case.